component/metrics/receiver.go

```diff
@@ -13,9 +14,51 @@ type Receiver struct {
 // RiverCapsule marks receivers as a capsule.
 func (r Receiver) RiverCapsule() {}
 
-// FlowMetric is a wrapper around a single metric without the timestamp
+// FlowMetric is a wrapper around a single metric without the timestamp.
 type FlowMetric struct {
-	GlobalRefID uint64
-	Labels      labels.Labels
-	Value       float64
+	globalRefID uint64
+	labels      labels.Labels
+	value       float64
+}
+
+// NewFlowMetric instantiates a new flow metric
+func NewFlowMetric(globalRefID uint64, lbls labels.Labels, value float64) *FlowMetric {
+	// Always ensure we have a valid global ref id
+	if globalRefID == 0 {
+		globalRefID = GlobalRefMapping.GetOrAddGlobalRefID(lbls)
+	}
+	return &FlowMetric{
+		globalRefID: globalRefID,
+		labels:      lbls,
+		value:       value,
+	}
+}
+
+// GlobalRefID Retrieves the GlobalRefID
+func (fw *FlowMetric) GlobalRefID() uint64 { return fw.globalRefID }
+
+// Value returns the value
+func (fw *FlowMetric) Value() float64 { return fw.value }
+
+// LabelsCopy returns a copy of the labels structure
+func (fw *FlowMetric) LabelsCopy() labels.Labels {
+	return fw.labels.Copy()
+}
```

Comment:
API nit: There's no reason to have a method for this, the caller can call `fw.RawLabels().Copy()` if they want to make a copy.

Reply:
I really wanted to bury RawLabels usage as much as possible since it's very easy to forget to add the copy.